woocommerce/woocommerce-ios
# [Orders with Coupons M4] Add discount to product - UI

Closes: #10201 


## Description

With this PR we implement the UI for adding a discount to a product on the Order Details. For that, we reused the existing UI components and classes for fees, whose logic was basically the same. We refactored those classes to include the discounts logic as well, using the prefix `FeeOrDiscountLineDetails`. This naming is not very generic but keeps clarity regarding its responsibility.

The discount business logic -calling the backend API and saving state-, and the UI for that state are not part of this PR.

## Testing instructions


1. Go to Orders
2. Create a new order
3. Add a product to the order
4. Once the product is added to the order, tap on the product
5. See that there is a new + row: "+ Add discount". Tap on it
6. See that we have a details screen to add discount in both percentage and fixed amount formats. Tap on Done or Close dismisses the screen.

Also:
- Please make sure that this functionality is hidden in the Release build configuration
- As we refactor components used by the Add Fee logic, make sure that it still works as expected

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/24bdb137-8e58-487d-9ef1-2145d4ade495

---
